### ISSUE TITLE
ssl/tls via websocket

### ISSUE THREAD

**Issue by bene40** [mqttjs/MQTT.js#1339]
II am trying to connect to a mqtt server with tls on websocket. I can with the client for macos MQTTX, with the attached configuration (note that I didn't have to specify the certificates), but I can't do it with mqttjs via browser, always the error related to the certificates. How it's possible?

<img width="662" alt="Schermata 2021-10-13 alle 14 33 28" src="https://user-images.githubusercontent.com/2782767/137133102-02341319-7844-450c-87e1-a8b5884ccc34.png">

**Comment by ridomin** [comment]
I'm having the exact same issue. Any progress?

Note that I can connect using secure web sockets from node, but not from the browser.

**Comment by DuyNT222054M** [comment]
I have same isssue, how can i fix it

**Comment by ZKUTF-8** [comment]
我有同样的问题，我该如何解决，有进展吗

**Comment by hsmptg** [comment]
Hi
I have a similar problem using mqttjs from inside a react app. When I use ws:// I can connect to the broker but if I use wss:// with self signed certificates I can not connect.
You could say that the problem is in my self hosted broker then, but if I preform the wss connection from a MQTT client I can connect to that broker!
Strangely, if in my react app I use the wss://broker.emqx.io:8084/mqtt it works again.
In conclusion it seems that mqttjs is able to connect using wss but only with some brokers and not with self hosted ones. I used a self hosted mosquitto and emqx broker with the same result.

**Comment by robertsLando** [comment]
TLS with certs doesn't work on browsers, only wss
